frontend/src/metabase/common/components/AggregationPicker/AggregationPicker.unit.spec.tsx

```diff
@@ -377,44 +372,47 @@ describe("AggregationPicker", () => {
 
     it("should allow picking a metric", () => {
       const metadata = createMetadata({ metrics: [TEST_METRIC] });
-      const { getRecentClause } = setupMetrics({ metadata });
+      const { getRecentClauseInfo } = setupMetrics({ metadata });
       const metric = checkNotNull(metadata.metric(TEST_METRIC.id));
 
       userEvent.click(screen.getByText(TEST_METRIC.name));
 
-      expect(getRecentClause()).toEqual(
-        expect.objectContaining({
-          displayName: metric.displayName(),
-        }),
-      );
+      expect(getRecentClauseInfo()).toMatchObject({
+        displayName: metric.displayName(),
+      });
     });
   });
 
   describe("custom expressions", () => {
     it("should allow to enter a custom expression", async () => {
-      const { onSelectLegacy } = setup();
+      const { getRecentClause, getRecentClauseInfo } = setup();
+
+      const expression = "1 + 1";
+      const expressionName = "My expression";
 
       userEvent.click(screen.getByText("Custom Expression"));
-      userEvent.type(screen.getByLabelText("Expression"), "1 + 1");
-      userEvent.type(screen.getByLabelText("Name"), "My expression");
+      userEvent.type(screen.getByLabelText("Expression"), expression);
+      userEvent.type(screen.getByLabelText("Name"), expressionName);
       userEvent.click(screen.getByRole("button", { name: "Done" }));
 
-      await waitFor(() =>
-        expect(onSelectLegacy).toHaveBeenCalledWith([
-          "aggregation-options",
-          ["+", 1, 1],
-          { "display-name": "My expression", name: "My expression" },
-        ]),
-      );
+      expect(getRecentClauseInfo()).toMatchObject({ displayName: expression });
+      expect(Lib.expressionName(getRecentClause())).toBe(expressionName);
     });
 
-    it("should open the editor when an expression is used", async () => {
+    it("should open the editor when a named expression without operator is used", async () => {
       setup({ query: createQueryWithInlineExpression() });
 
       expect(screen.getByText("Custom Expression")).toBeInTheDocument();
       expect(screen.getByDisplayValue("Avg Q")).toBeInTheDocument();
     });
 
+    it("should open the editor when a named expression with operator is used", async () => {
```

Comment:
I added this to verify 2nd bullet point from this [Slack message](https://metaboat.slack.com/archives/C04CYTEL9N2/p1701257788565959?thread_ts=1701083796.853609&cid=C04CYTEL9N2). It's currently failing.

I'm not sure about the test name though, suggestions will be appreciated.